core/src/main/java/com/sequenceiq/cloudbreak/service/proxy/ProxyConfigDtoService.java

```diff
@@ -0,0 +1,69 @@
+package com.sequenceiq.cloudbreak.service.proxy;
+
+import javax.inject.Inject;
+import javax.ws.rs.WebApplicationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.vault.VaultException;
+
+import com.sequenceiq.cloudbreak.dto.ProxyConfig;
+import com.sequenceiq.cloudbreak.service.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.service.secret.model.SecretResponse;
+import com.sequenceiq.cloudbreak.service.secret.service.SecretService;
+import com.sequenceiq.environment.api.v1.proxy.endpoint.ProxyEndpoint;
+import com.sequenceiq.environment.api.v1.proxy.model.response.ProxyResponse;
+import com.sequenceiq.environment.client.EnvironmentServiceClient;
+
+@Service
+public class ProxyConfigDtoService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProxyConfigDtoService.class);
+
+    @Inject
+    private EnvironmentServiceClient environmentServiceClient;
```

Comment:
Simply injecting the `ProxyEndpoint` class would be an easier way to communicate with the Env service

Reply:
That will be implemented in an individual PR, as DistroX Api related converters will require more modification.